packages/mongodb-rag-core/src/DataStreamer.ts

```diff
@@ -135,11 +136,10 @@ export function makeDataStreamer(): DataStreamer {
         throw new Error("Tried to connect SSE, but it was already connected.");
       }
       sse = makeServerSentEventDispatcher<SomeStreamEvent>(res);
-      // If the client closes the connection, stop sending events
       res.on("close", () => {
-        if (this.connected) {
-          this.disconnect();
-        }
+        logger.info(
+          "Client closed the streaming connection before the stream was finished."
+        );
```

Comment:
does it disconnect automatically now?

Reply:
`disconnect()` is a wrapper around `res.end()` which completes the request on the client.

This is actually a bit of a bug - if we get a `close` event then the client has already disconnected so it's too late to end the request.